src/mapeo-project.js

```diff
@@ -113,6 +115,25 @@ export class MapeoProject {
         db,
       }),
     }
+
+    this.#blobStore = new BlobStore({
+      coreManager: this.#coreManager,
+    })
+
+    this.$blobs = {
+      /**
+       * 
+       * @param {import('./types.js').BlobId} blobId 
+       * @returns {String}
+       */
+      getUrl: (blobId) => {
+        const { driveId, type, variant, name } = blobId
+        // TODO: where is the hostname set?
+        // TODO: expose the projectId
```

Comment:
> TODO: where is the hostname set?

My guess is that we'd want to pass an opt to the `MapeoProject` constructor to specify the hostname. In general, I think it makes sense for the hostname to be generated by whatever is creating the `BlobServer` instance e.g. at the app level. Currently the `BlobServer` doesn't [handle multiprojects](https://github.com/digidem/mapeo-core-next/blob/52e6f12b9a16473ce6e3fcc85e716570cbfa2385/src/blob-server/index.js#L11) but that's the goal for it so that only one instance of a server is created for the entire application. 

> TODO: expose the projectId

think it has to be exposed based on [how the `BlobServer` is implemented](https://github.com/digidem/mapeo-core-next/blob/52e6f12b9a16473ce6e3fcc85e716570cbfa2385/src/blob-server/fastify-plugin.js#L60)

Reply:
For port we'll need to expose a method on blobServer like `getPort()` - should be async and return once the server has started (it will need to close every time the app goes into the background). We can use that to ensure that getUrl only resolves when the blob server is actually running.

Reply:
You should be able to do:

```ts
async function getPort(server: import('node:http').Server) {
  const addr = server.address()
  const port = addr && typeof addr === 'object' ? addr.port : null
  if (typeof port = 'number') return port
  await once(server, 'listening')
  return getPort(server)
}
```

You can get `server` from [`fastify.server()`](https://fastify.dev/docs/latest/Reference/Server#server)

No strong opinions on whether we pass down the fastify instance, http.server instance, or just the getPort function.